### ISSUE TITLE
[E2E] HF/Timm/Torchbench models got "eager_two_runs_differ"

### ISSUE THREAD

**Issue by libohao1201** [intel/torch-xpu-ops#1256]
### 🐛 Describe the bug

The following models got "eager_two_runs_differ"
|     |            |                                                                                                           eager_two_runs_diff                                                                                                          |
|-----|------------|:--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------:|
| LNL | HF         |                                                                                                                                                                                                                                        |
|     | Timm       |                                                                                                                                                                                                                                        |
|     | Torchbench | Super_SloMo   (train_eager_fp32/amp_bf16)      pytorch_CycleGAN_and_pix2pix (train_eager_fp32/bf16/amp_bf16)                                                                                                                           |
| BMG | HF         |                                                                                                                                                                                                                                        |
|     | Timm       |                                                                                                                                                                                                                                        |
|     | Torchbench | Super_SloMo   (train_eager_fp32)      pytorch_CycleGAN_and_pix2pix (train_eager_fp32/bf16)                                                                                                                                             |
| ARC | HF         | DistilBertForMaskedLM(train_fp16_eager)                                                                                                                                                                                                |
|     | Timm       | convnext_base   (train_eager)      jx_nest_base (train_eager)      swin_base_patch4_window7_224 (train_eager)      twins_pcpvt_base (train_eager)      coat_lite_mini (train)      convit_base      mobilevit_s      tnt_s_patch16_224 |
|     | Torchbench | hf_Reformer(train_eager)      timm_regnet (train_eager_fp32)       

### Versions

Env:
| stock pytorch  | https://github.com/pytorch/pytorch/tree/90b7dcf2c5ee13b892701822f2abbc0e64f5584d pip install --pre torch==2.6.0.dev20241202+xpu torchvision==0.20.0.dev20241202+xpu torchaudio==2.5.0.dev20241202+xpu --index-url https://download.pytorch.org/whl/nightly/xpu |   |
|----------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|:-:|
| torch-xpu-ops  | Commit: bf4bab1ffffd84e5f747f65a17e08ee2fe633102 Commit: 0f48ac07e42ce30d2d07447f4b49bb4ab23f8e64   (including https://github.com/intel/torch-xpu-ops/pull/1187)  - for UT                                                                                     |   |
| Driver         | 32.0.101.6314  32.0.101.6252(bmg)                                                                                                                                                                                                                              |   |
| Conda          | python 3.10                                                                                                                                                                                                                                                    |   |
| transformer    | 243e186efbf7fb93328dd6b34927a4e8c8f24395

**Comment by mengfei25** [comment]
Same on PVC for FP32 training

**Comment by jianyizh** [comment]
For Super_SloMo and pytorch_CycleGAN_and_pix2pix, pytorch did not set deterministic, see
 https://github.com/pytorch/pytorch/blob/2a7e997b3f1805b077810d7fef87cabc4411eeea/benchmarks/dynamo/common.py#L3582

pytorch_CycleGAN_and_pix2pix is reasonable to fail on eager-two-run-diff because of atomic ops. 

Super_SloMo can get pass if I use torch.use_deterministic_algorithms(True,warn_only=True) and https://github.com/intel/torch-xpu-ops/pull/1370 to avoid atomic ops.

**Comment by jianyizh** [comment]
hf_Reformer, swin_base_patch4_window7_224, twins_pcpvt_base,  coat_lite_mini, mobilevit_s, tnt_s_patch16_224, can pass in latest pvc weekly. It's strange that I don't see  convit_base in weekly, but I can pass on local pvc. @Stonepia  please check on LNL and ARC

**Comment by jianyizh** [comment]
Let's use https://github.com/intel/torch-xpu-ops/issues/1274 to check convnext_base  and https://github.com/intel/torch-xpu-ops/issues/979 to check jx_nest_base on PVC. @Stonepia please note that for these two models,  they should be fail_accuracy on PVC, but you get eager_two_runs_diff on ARC, mean there are extra accuracy issues on ARC so that these models fails earlier... please have a check

**Comment by Stonepia** [comment]
We still get quite a few eager two run differ on BMG / LNL

```
# FP32 Training
Super_SloMo   
demucs
pytorch_CycleGAN_and_pix2pix 
```

So those ops are not fixed yet. I would think this is the same reason as PVC.

**Comment by Stonepia** [comment]
> For Super_SloMo and pytorch_CycleGAN_and_pix2pix, pytorch did not set deterministic, see https://github.com/pytorch/pytorch/blob/2a7e997b3f1805b077810d7fef87cabc4411eeea/benchmarks/dynamo/common.py#L3582
> 
> pytorch_CycleGAN_and_pix2pix is reasonable to fail on eager-two-run-diff because of atomic ops.
> 
> Super_SloMo can get pass if I use torch.use_deterministic_algorithms(True,warn_only=True) and [#1370](https://github.com/intel/torch-xpu-ops/pull/1370) to avoid atomic ops.

I think those should have already used deterministic as in the following code, so I assume the only reason is because we have issues in the atomic op?

https://github.com/pytorch/pytorch/blob/924a247fbbe32063966533535b1cc0cf474ef1ed/benchmarks/dynamo/common.py#L3577-L3593

**Comment by jianyizh** [comment]
> > For Super_SloMo and pytorch_CycleGAN_and_pix2pix, pytorch did not set deterministic, see https://github.com/pytorch/pytorch/blob/2a7e997b3f1805b077810d7fef87cabc4411eeea/benchmarks/dynamo/common.py#L3582
> > pytorch_CycleGAN_and_pix2pix is reasonable to fail on eager-two-run-diff because of atomic ops.
> > Super_SloMo can get pass if I use torch.use_deterministic_algorithms(True,warn_only=True) and [#1370](https://github.com/intel/torch-xpu-ops/pull/1370) to avoid atomic ops.
> 
> I think those should have already used deterministic as in the following code, so I assume the only reason is because we have issues in the atomic op?
> 
> https://github.com/pytorch/pytorch/blob/924a247fbbe32063966533535b1cc0cf474ef1ed/benchmarks/dynamo/common.py#L3577-L3593

No, only models not in the list will set deterministic, otherwise it will have error. And atomic op will cause non-deterministic behavior

**Comment by jianyizh** [comment]
We need check the failed models on arc. If it's not important, we can check it on pytorch 2.8

**Comment by chuanqi129** [comment]
BMG & LNL have common model list, prioritize them
